libs/language-server/src/lib/ast/expressions/type-inference.ts

```diff
@@ -325,9 +327,13 @@ function inferTypeFromReferenceLiteral(
   if (isTransformDefinition(referenced)) {
     return PrimitiveValuetypes.Transform;
   }
-  if (isTransformPortDefinition(referenced)) {
-    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
-    const valueType = referenced?.valueType;
+  if (
+    isTransformPortDefinition(referenced) ||
+    isBlocktypeProperty(referenced)
+  ) {
+    const valueType = isTransformPortDefinition(referenced)
+      ? referenced.valueType
+      : referenced.valuetype;
```

Comment:
Inconsistency `valuetype` vs. `valueType`. I believe we used camel case for value types in existing code. That should also make a uniform access possible here.